src/semeio/fmudesign/design_distributions.py

```diff
@@ -513,37 +514,28 @@ def is_number(teststring):
         return False
 
 
-def read_correlations(corr_dict, corrsheet):
+def read_correlations(excel_filename: Path, corr_sheet: str) -> pd.DataFrame:
     """Reading correlation info for a
     monte carlo sensitivity
 
     Args:
-        corr_dict (OrderedDict): correlation info
+        excel_filename (Path): Path to Excel file containing correlation matrix
+        corr_sheet (str): name of sheet containing correlation matrix
 
     Returns:
         pd.DataFrame: Dataframe with correlations, parameter names
             as column and index
     """
-    correlations = None
-    filename = corr_dict["inputfile"]
-    if corrsheet in corr_dict["sheetnames"]:
-        if str(filename).endswith(".xlsx"):
-            correlations = pd.read_excel(
-                filename, corrsheet, index_col=0, engine="openpyxl"
-            )
-            correlations.dropna(axis=0, how="all", inplace=True)
-            correlations = correlations.loc[
-                :, ~correlations.columns.str.contains("^Unnamed")
-            ]
-        else:
-            raise ValueError(
-                "Correlation matrix filename should be on "
-                "Excel format and end with .xlsx "
-            )
-    else:
+    if not str(excel_filename).endswith(".xlsx"):
```

Comment:
More canonical would be `if excel_filename.suffix != ".xlsx"`

Reply:
Thanks, but that causes a test to fail. I've fixed type hint to indicate that this can be both a str and Path.
Long way to go before this code-base is up to standard, but it's a start :)